staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/monitoring.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversion
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	latencyBuckets = prometheus.ExponentialBuckets(0.001, 2, 15)
+)
+
+// converterMonitoringFactory holds metrics for all CRD converters
+type converterMonitoringFactory struct {
+	durations   map[string]*prometheus.HistogramVec
+	factoryLock sync.Mutex
+}
+
+func newConverterMonitoringFactory() *converterMonitoringFactory {
+	return &converterMonitoringFactory{durations: map[string]*prometheus.HistogramVec{}, factoryLock: sync.Mutex{}}
+}
+
+var _ crConverterInterface = &converterMonitoring{}
+
+type converterMonitoring struct {
+	delegate  crConverterInterface
+	latencies *prometheus.HistogramVec
+	labels    []string
+	crdName   string
+}
+
+func (c *converterMonitoringFactory) monitor(converterName string, crdName string, converter crConverterInterface) (crConverterInterface, error) {
+	c.factoryLock.Lock()
+	defer c.factoryLock.Unlock()
+	metric, exists := c.durations[converterName]
+	if !exists {
+		metric = prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Namespace: "apiserver",
+				Subsystem: "crd",
```

Comment:
this will result in `apiserver_crd_crd...`, typically the `.Namespace` and `.Subsystem` field is entirely skipped and just included in the `.Name` directly

Reply:
fixed